proto/rdp/reviewdog.proto

```diff
@@ -0,0 +1,192 @@
+// Reviewdog Diagnostic Protocol (RDP)
+//
+// Reviewdog Diagnostic Protocol (RDP. Or *R*eview*D*og *P*rotocol in short.)
+// defines generic machine readable message structures which represents a
+// result of diagnostic tool such as a compiler or a linter.
+//
+// The idea behind the Reviewdog Diagnostic Protocol (RDP) is to standardize
+// the protocol for how diagnostic tools (e.g. compilers, linters, etc..) and
+// development tools (e.g. editors, reviewdog, etc..) communicate.
+//
+// Wire formats of Reviewdog Diagnostic Protocol.
+// - rdpjsonl: JSON Lines (http://jsonlines.org/) of the `Diagnostic` message.
+// - rdpjson: JSON format of the `DiagnosticResult` message.
+
+syntax = "proto3";
```

Comment:
I think using a JSON Schema file to describe the format would be easier for others to use and validate their output

Reply:
I prefer Protocol Buffer (as primary IDL) because...

- Protobur can auto-generate code for a lot of languages including Go which is used in reviewdog.
- It's more readable (I believe) and I think it's actually not difficult to understand. It supports comments for example.

But, yes that's a good point. 
It should be able to auto-generate json schema from protobuf so we can give it a try. https://github.com/chrusty/protoc-gen-jsonschema

Reply:
Added. 3e7cb51d7ea1a608c0869052054231e3c72ffade